src/test/controllers/controller_s4mk3_settings_test.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ This test case is used to ensure that hardcoded CO value in the the settings
+ definition matches with Mixxx value and will help detecting regression if they
+ are ever updated.
+*/
+#include "controllers/legacycontrollermapping.h"
+#include "controllers/legacycontrollermappingfilehandler.h"
+#include "library/trackmodel.h"
+#include "test/mixxxtest.h"
+#include "util/time.h"
+
+class ControllerS4MK3SettingTest : public MixxxTest {
+  protected:
+    void SetUp() override {
+        mixxx::Time::setTestMode(true);
+        mixxx::Time::setTestElapsedTime(mixxx::Duration::fromMillis(10));
+    }
+
+    void TearDown() override {
+        mixxx::Time::setTestMode(false);
+    }
+
+    static QHash<QString, TrackModel::SortColumnId> COLUMN_MAPPING;
+};
+
+QHash<QString, TrackModel::SortColumnId> ControllerS4MK3SettingTest::COLUMN_MAPPING = {
+        {"Artist", TrackModel::SortColumnId::Artist},
```

Comment:
It is unfortunate that we don't have enum-to-string mapping in the actual code. https://github.com/mixxxdj/mixxx/blob/main/src/library/browse/browsetablemodel.cpp#L60

There is an issue with internationalization, we will need a solution for that too eventually.  But for now this is good enough, thank you!

Reply:
Maybe add a comment to this file that if other controllers want to add the same type of feature they can also use this test?  Maybe rename the test to columnid_regression_test.cpp or something like that?

Reply:
Done!